### PR TITLE
Add Tween.startAt(time) API and add more unit test.

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #17078

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/tween/actions/action-interval.ts [hunk 1]
```diff
@@ -80,6 +82,11 @@ export abstract class ActionInterval extends FiniteTimeAction {
         }
     }
 
+    setStartTime (time: number): void {
```
Comment (minggo) [inline cocos/tween/actions/action-interval.ts:84]: Does it support negative values?

Comment (dumganhar) [reply cocos/tween/actions/action-interval.ts:84]: I think it will not, I will add a valid time checking in this function.

Comment (minggo) [reply cocos/tween/actions/action-interval.ts:84]: If so, then i think don't have to add a variable to check if it is set or not, just check the value. The default value can be set to a negative value.

Comment (dumganhar) [reply cocos/tween/actions/action-interval.ts:84]: Updated and added two unit tests for this.

Comment (dumganhar) [reply cocos/tween/actions/action-interval.ts:84]: > If so, then i think don't have to add a variable to check if it is set or not, just check the value. The default value can be set to a negative value.

Good point. I think the default value could be 0, I will update the code.

Comment (dumganhar) [reply cocos/tween/actions/action-interval.ts:84]: Updated

File: cocos/tween/actions/action-interval.ts [hunk 2]
```diff
@@ -118,7 +124,13 @@ export abstract class ActionInterval extends FiniteTimeAction {
         dt *= this._speed;
         if (this._firstTick) {
             this._firstTick = false;
-            this._elapsed = 0;
+            let elapsed = 0;
```
Comment (minggo) [inline cocos/tween/actions/action-interval.ts]: It seems this variable is not needed.

Comment (dumganhar) [reply cocos/tween/actions/action-interval.ts]: Fixed.

**Review by minggo — APPROVED** [review]

**Pull request merged**